estimagic/optimization/nlopt_optimizers.py

```diff
@@ -141,6 +140,899 @@ def nlopt_neldermead(
+    criterion_and_derivative,
+    x,
+    lower_bounds,
+    upper_bounds,
+    *,
+    convergence_relative_params_tolerance=CONVERGENCE_RELATIVE_PARAMS_TOLERANCE,
+    convergence_absolute_params_tolerance=CONVERGENCE_ABSOLUTE_PARAMS_TOLERANCE,
+    convergence_relative_criterion_tolerance=CONVERGENCE_RELATIVE_CRITERION_TOLERANCE,
+    convergence_absolute_criterion_tolerance=CONVERGENCE_ABSOLUTE_CRITERION_TOLERANCE,
+    stopping_max_criterion_evaluations=STOPPING_MAX_CRITERION_EVALUATIONS_GLOBAL,
+):
+    """Optimize a scalar function using the ISRES algorithm.
+
+    ISRES is an implementation of "Improved Stochastic Evolution Strategy" written
+    for solving optimization problems with non-linear constraints. The algorithm
+    is supposed to be a global method, in that it has heuristics to avoid local
+    minima. However, no convergence proof is available.
+
+    The original method and a refined version can be found, respecively, in:
+    Thomas Philip Runarsson and Xin Yao, "Search biases in constrained evolutionary
+    optimization," IEEE Trans. on Systems, Man, and Cybernetics Part C: Applications
+    and Reviews, vol. 35 (no. 2), pp. 233-243 (2005).
+    Thomas P. Runarsson and Xin Yao, "Stochastic ranking for constrained evolutionary
+    optimization," IEEE Trans. Evolutionary Computation, vol. 4 (no. 3), pp. 284-294
+    (2000).
+
+
+    ``nlopt_isres`` supports the following ``algo_options``:
+
+    - convergence.relative_params_tolerance (float):  Stop when the relative movement
+      between parameter vectors is smaller than this.
+    - convergence.relative_criterion_tolerance (float): Stop when the relative
+      improvement between two iterations is smaller than this.
+    - stopping_max_criterion_evaluations_global (int): If the maximum number of function
+      evaluation is reached.
+
+
+    """
+    out = _minimize_nlopt(
+        criterion_and_derivative,
+        x,
+        lower_bounds,
+        upper_bounds,
+        algorithm=nlopt.GN_ISRES,
+        algorithm_name="nlopt_isres",
+        convergence_xtol_rel=convergence_relative_params_tolerance,
+        convergence_xtol_abs=convergence_absolute_params_tolerance,
+        convergence_ftol_rel=convergence_relative_criterion_tolerance,
+        convergence_ftol_abs=convergence_absolute_criterion_tolerance,
+        stopping_max_eval=stopping_max_criterion_evaluations,
+    )
+    return out
+
+
+def nlopt_crs2_lm(
+    criterion_and_derivative,
+    x,
+    lower_bounds,
+    upper_bounds,
+    *,
+    convergence_relative_params_tolerance=CONVERGENCE_RELATIVE_PARAMS_TOLERANCE,
+    convergence_absolute_params_tolerance=CONVERGENCE_ABSOLUTE_PARAMS_TOLERANCE,
+    convergence_relative_criterion_tolerance=CONVERGENCE_RELATIVE_CRITERION_TOLERANCE,
+    convergence_absolute_criterion_tolerance=CONVERGENCE_ABSOLUTE_CRITERION_TOLERANCE,
+    stopping_max_criterion_evaluations=STOPPING_MAX_CRITERION_EVALUATIONS_GLOBAL,
+    random_search_population_size=None,
```

Comment:
We probably want to rename this to `genetic_popsize`. @roecla  is that the name you are using in pygmo? Maybe we also already have a way of calculating the default there and it would be nice if they are harmonized. 

I suggest that we just merge this PR first and then @roecla  makes sure that genetic algorithms from pygmo and nlopt have aligned names and default values.